h2o-core/src/main/java/hex/schemas/GridSearchSchema.java

```diff
@@ -122,6 +126,24 @@
       parms.remove("grid_id");
     }
 
+    if (parms.containsKey("parallelism_level")) {
+      final String parallelismLevelProperty = parms.getProperty("parallelism_level");
+      try {
+        this.parallelism_level = Integer.parseInt(parallelismLevelProperty);
+        if (this.parallelism_level < 0) {
+          throw new IllegalArgumentException(String.format("Parallelism level must be >= 0. Given value: '%d'",
+                  parallelism_level));
+        }
+      } catch (NumberFormatException e) {
+        final String errorMessage = String.format("Could not parse given parallelism_level value: '%s' - not a number.",
+                parallelismLevelProperty);
+        throw new IllegalArgumentException(errorMessage, e);
+      }
+      parms.remove("parallelism_level");
+    } else {
+      this.parallelism_level = 0; // Operate in sequential mode by default
+    }
+
```

Comment:
Waw! Why is this impl of `fillFromParms` so complicated in this class? Why is it not done automatically like for other schemas? Someone needs to fix this.
Look at how complicated it is just to set a `int`.
With auto-binding, you would just have to check that the bound value is `>= 0`...

Reply:
No idea, ask the original author. Refactoring this is not part of this PR. But you're always welcome to create a PR with a fix.